File: 2nd Year/2nd Semester/MPP/PROIECT-JAVA/Laborator/TriatlonNetworking/src/main/java/triatlon/network/jsonprotocol/Request.java
package triatlon.network.jsonprotocol;


import triatlon.network.dto.*;

public class Request {
    private RequestType type;
    private ArbitruDTO arbitruDTO;
    private ParticipantDTO participantDTO;
    private RezultatDTO rezultatDTO;
    private ProbaDTO probaDTO;
    private TipProbaDTO tipProbaDTO;

    public Request() {
    }

    public RequestType getType() {
        return type;
    }

    public void setType(RequestType type) {
        this.type = type;
    }

    public ArbitruDTO getArbitruDTO() {
        return arbitruDTO;
    }

    public void setArbitruDTO(ArbitruDTO arbitruDTO) {
        this.arbitruDTO = arbitruDTO;
    }

    public ParticipantDTO getParticipantDTO() {
        return participantDTO;
    }

    public void setParticipantDTO(ParticipantDTO participantDTO) {
        this.participantDTO = participantDTO;
    }

    public RezultatDTO getRezultatDTO() {
        return rezultatDTO;
    }

    public void setRezultatDTO(RezultatDTO rezultatDTO) {
        this.rezultatDTO = rezultatDTO;
    }

    public ProbaDTO getProbaDTO() {
        return probaDTO;
    }

    public void setProbaDTO(ProbaDTO probaDTO) {
        this.probaDTO = probaDTO;
    }

    @Override
    public String toString() {
        return "Request{" +
                "type=" + type +
                ", arbitruDTO=" + arbitruDTO +
                ", participantDTO=" + participantDTO +
                ", rezultatDTO=" + rezultatDTO +
                ", probaDTO=" + probaDTO +
                '}';
    }

    public TipProbaDTO getTipProbaDTO() {
        return tipProbaDTO;
    }

    public void setTipProbaDTO(TipProbaDTO tipProbaDTO) {
        this.tipProbaDTO = tipProbaDTO;
    }
}

File: 2nd Year/2nd Semester/MPP/PROIECT-C#/Lab1/TriatlonNetworking/jsonprotocol/Request.cs
using TriatlonNetworking.dto;

namespace TriatlonNetworking.jsonprotocol;

public class Request
{
    public RequestType Type { get; set; }
    public ArbitruDTO ArbitruDTO { get; set; }
    public ParticipantDTO ParticipantDTO { get; set; }
    public RezultatDTO RezultatDTO { get; set; }
    public ProbaDTO ProbaDTO { get; set; }
    public TipProbaDTO TipProbaDTO { get; set; }

    public Request() { }

    public override string ToString()
    {
        return $"Request{{type={Type}, arbitruDTO={ArbitruDTO}, participantDTO={ParticipantDTO}, rezultatDTO={RezultatDTO}, probaDTO={ProbaDTO}, tipProbaDTO={TipProbaDTO}}}";
    }
}